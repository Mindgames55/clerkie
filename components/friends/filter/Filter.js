'use client'
import { useState } from 'react'
import Hamburger from '../../../public/hamburger.svg'
import styles from './Filter.module.css'
import Clear from './Clear'
import Dropdown from './Dropdown'

export default function Filter({filters, submitFilters, numberOfActiveFilters, clearFilters}) {
    const [isDropdownVisible, setIsDropDownVisible] = useState(false);

    function handleDropdownToggleVisibility() {
        setIsDropDownVisible(!isDropdownVisible)
    }

    const isFilterButtonActive = isDropdownVisible || numberOfActiveFilters > 0;
    return (
        <div className={styles.container}>
            <div className={styles.controls}>
                <button className={isFilterButtonActive ? styles['active-filter'] : styles.filter}
                        aria-label='filter by status'
                        onClick={handleDropdownToggleVisibility}
                        >
                            <Hamburger />
                            {numberOfActiveFilters > 0 && <span className={styles['filter-number']}>{numberOfActiveFilters}</span>}
                </button>
                <span className={styles.separator}>|</span>
                <Clear handleClick={clearFilters} isDisabled={numberOfActiveFilters === 0}/>
            </div>
            <Dropdown isVisible={isDropdownVisible} toggleVisibility={handleDropdownToggleVisibility} filters={filters} submitFilters={submitFilters} numberOfActiveFilters={numberOfActiveFilters} />
        </div>
    )
}