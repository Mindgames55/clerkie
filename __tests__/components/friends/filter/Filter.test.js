import { render, screen } from '@testing-library/react';
import Filter from '@/components/friends/filter/Filter'

export const props = {
    filters: [{
        filter: 'status',
        value: 'close',
        applied: false
    }],
    numberOfActiveFilters: 3
}

it('renders Filter correctly', () => {
    const { container } = render(<Filter {...props} />);
    expect(container).toMatchSnapshot();
});