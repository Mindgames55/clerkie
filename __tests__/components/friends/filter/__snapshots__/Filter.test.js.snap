// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Filter correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="controls"
    >
      <button
        aria-label="filter by status"
        class="active-filter"
      >
        <span />
        <span
          class="filter-number"
        >
          3
        </span>
      </button>
      <span
        class="separator"
      >
        |
      </span>
      <button
        class="clear"
      >
        Clear all
      </button>
    </div>
    <div
      class="container hidden"
    >
      <header
        class="header"
      >
        <div
          class="box"
        >
          <button
            class="clear"
            disabled=""
          >
            Clear all
          </button>
        </div>
        <div
          class="box"
        >
          <span
            class="title"
          >
            Filter
          </span>
        </div>
        <div
          class="box"
        >
          <button
            aria-label="close"
          >
            <span />
          </button>
        </div>
      </header>
      <div
        class="filter-container"
      >
        <p
          class="status-title"
        >
          Friend Status
        </p>
        <ul
          class="options"
        >
          <li
            class="option"
          >
            <label
              class="option-copy"
              for="close"
            >
              Close Friends
            </label>
            <input
              class="checkbox"
              id="close"
              name="close"
              type="checkbox"
            />
          </li>
        </ul>
        <button
          class="apply"
        >
          Apply
        </button>
      </div>
    </div>
  </div>
</div>
`;
